src/main/kotlin/com/desmonddavid/bots/digitforumbot/bots/NewPostsBot.kt

```diff
@@ -132,7 +137,36 @@ object NewPostsBot : TelegramLongPollingBot() {
                 }\n\n"
             message += entry
         }
-
+        if(checkIfNew(message) ==1){
+            message=""
+        }
         return message
     }
+
+    private fun checkIfNew(message: String): Int {
+
+        var lastHash = ""
+
+        val hashable = message.toByteArray(Charsets.UTF_8)
+        val md = MessageDigest.getInstance("MD5")
+        val newHash = String.format("%032x", BigInteger(1, md.digest(hashable)))
+
+        // Check if file exists, if it does not exist, create it.
+        val lastHashFile = File(LAST_HASH_FILENAME)
+        if (!lastHashFile.exists()) {
```

Comment:
Perform file operations before calculating hash. In other words, move this block to the top of the method.